src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java

```diff
@@ -338,6 +339,12 @@ private static long getGroupKey(MarkDuplicatesSparkRecord record) {
         }
```

Comment:
Are you checking the sequence dictionary on startup to make sure that we don't have > 64k contigs?

Reply:
Can this method be made an instance method on MarkDuplicatesSparkRecord?

Reply:
I added some checks for libraries and contigs so an exception is thrown in the case where we expect the keys to break.